enhancements/network/bgp-overview.md

```diff
@@ -0,0 +1,257 @@
+multiple service instances. Mobile clients may be expected to roam between
+endpoints as the number of hops changes with location, so this is suited to
+either fully stateless services over UDP such as DNS or video streaming, or
+semi-stateless services over TCP such as RESTful API endpoints. This technique
+could be used to load balance pods or services directly, or to expose multiple
+LoadBalancer instances.
+
+### Virtual Network Interconnect and Hybrid Cloud
+
+The Pod network is by default an isolated virtual network.  There is sometimes
+a desire to interconnect multiple virtual networks together.  This could be
+between multiple Kubernetes clusters, but it doesn't have to be the case.
+
+BGP, specifically MP-BGP, is commonly used as a signaling protocol, together
+with other data planes technologies like MPLS and VXLAN to interconnect virtual
+network at L2 or L3.
+
+BGP EVPN (Ethernet VPN, or Virtual Private Network) is a BGP technology that is
+discussed in the context of this use case.  BGP EVPN can be used to advertise
+MAC addresses and IP-MAC address bindings.  BGP EVPN is not a full solution to
+this use case by itself, as it is a technology that can help implement part of
+the control plane.  There are more details that must be explored, including
+what the data plane looks like between the two clusters.
+
+BGP and MPLS is heavily used in the Telco area, and by the Cloud providers to
+interconnect their facilities to the Cloud, i.e. [AWS Direct
+connect](https://d1.awsstatic.com/whitepapers/Networking/integrating-aws-with-multiprotocol-label-switching.pdf).
+
+This is a significantly non-trivial effort.  Future enhancements are needed to
+discuss this in much more detail.
+
+## Common Design Considerations
+
+This section summarizes the common design considerations for OpenShift in
+support of BGP related features.
+
+### Avoid L2 (Layer 2) Domain Assumptions
+
+We must avoid the assumption of Nodes residing on the same L2 domain as much as
+possible.  Every OpenShift Node should only assume L3 connectivity to every
+other Node.
+
+One example that should be avoided is any use of keepalived, as all Nodes
+running keepalived to manage a VIP (Virtual IP address) must reside on the same
+L2 segment since ARP (IPv4) or NDP (IPv6) is used to announce the new location
+of the VIP when it moves to a new Node.
+
+### Avoid any Hard Requirements on BGP
+
+Any use of BGP in OpenShift should remain optional.  We have some customers
+that do not want to use BGP at all, either for some technical or policy reason.
+We may explore using it to add optional features, but it should not be required
+for any OpenShift clusters.
+
+For example, we are considering [adding MetalLB][3] to OpenShift to support
+Services of `type=LoadBalancer`.  While MetalLB can use BGP, it also has an
+alternative `layer2` mode that can provide the same functionality, though with
+a different set of limitations.
```

Comment:
We could also document how customers can set up their routing manually to route the Cluster IP range to the worker nodes instead of doing metallb for layer 2, right?

Reply:
I've added another sentence here to point out that static routing is another alternative to avoid the BGP dependency